### PR TITLE
Refactor Builder Plans to leverage `pkg_bind`/`pkg_export`

### PR DESCRIPTION
**Pull request by reset**
All of the builder services are currently leveraging the `ParseInto` trait to read data directly from a `config.toml` written by the Supervisor to the service directory. We provide no contract for that file - it's not meant to be read by applications - it's specifically for the Supervisor.

Additionally, we are not leveraging `pkg_bind` or `pkg_export` to automatically discover required services.

* Refactor config modules to read from a standard toml file tempalted by Habitat's Supervisor
* Refactor builder plans to leverage `pkg_bind`/`pkg_export`
* Remove unused plans

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by georgemarshall** [comment]
It might be useful to have some ipv6 addresses in the test cases.

File: components/builder-api/src/config.rs [hunk 1]
```diff
@@ -15,115 +15,164 @@
...
+        heartbeat = 9001
+
+        [github]
+        url = "https://api.github.com"
+        client_id = "0c2f738a7d0bd300de10"
+        client_secret = "438223113eeb6e7edf2d2f91a232b72de72b9bdf"
+        "#;
+
+        let config = Config::from_raw(&content).unwrap();
+        assert_eq!(&format!("{}", config.http.listen), "0:0:0:0:0:0:0:1");
```
Comment (mwrock) [inline components/builder-api/src/config.rs]: change to `"::1"` to fix failing test

**Comment by mwrock** [comment]
Just needs a trivial fix to get this green, It looks good to me.

**Comment by reset** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/224986352).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/224986352).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**